src/strategies/convex/strategy-convex-stecrv.sol

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.2;
+
+import "../../lib/erc20.sol";
+import "../../lib/safe-math.sol";
+
+import "../../interfaces/jar.sol";
+import "../../interfaces/steth.sol";
+import "../../interfaces/weth.sol";
+import "../../interfaces/curve.sol";
+import "../../interfaces/uniswapv2.sol";
+import "../../interfaces/controller.sol";
+
+import "../strategy-base.sol";
+
+interface ICurveFi {
+    function add_liquidity(
+        // stETH pool
+        uint256[2] calldata amounts,
+        uint256 min_mint_amount
+    ) external payable;
+
+    function balances(int128) external view returns (uint256);
+}
+
+interface IConvexDepositor {
+    function deposit(uint256 _pid, uint256 _amount, bool _stake) external returns(bool);
+    function depositAll(uint256 _pid, bool _stake) external returns(bool);
+}
+
+interface IConvexRewards {
+    //get balance of an address
+    function balanceOf(address _account) external view returns(uint256);
+    // primary reward balance (will be in CRV)
+    function earned(address account) external view returns (uint256);
+    // extra rewards array getter
+    function extraRewards(uint256) external view returns (address);
+    function extraRewardsLength() external view returns (uint256);
+
+    //withdraw directly to curve LP token
+    function withdrawAndUnwrap(uint256 _amount, bool _claim) external returns(bool);
+    //claim rewards
+    function getReward() external returns(bool);
+}
+
+interface IConvexExtraRewards {
+    function earned(address account) external view returns (uint256);
+}
+
+interface IConvexToken {
+    function maxSupply() external view returns (uint256);
+    function reductionPerCliff() external view returns (uint256);
+    function totalCliffs() external view returns (uint256);
+    function totalSupply() external view returns (uint256);
+}
```

Comment:
can you make the interfaces in a separate file? we have an `interface` folder.

Reply:
Replaced these definitions by importing strategy-convex-farm-base.sol and interfaces/steth.sol. The payable add_liquidity function is defined in ICurveFi_4 in interfaces/curve.sol